aws/credentials/ssocreds/provider.go

```diff
@@ -64,9 +65,12 @@ type Provider struct {
 	// parameter will be ignored.
 	CachedTokenFilepath string
 
+	// Used by the SSOCredentialProvider to judge if TokenProvider is configured
+	HasTokenProvider bool
```

Comment:
Why do we need this? `bearer.TokenProvider` is an interface, you can check if it's `nil` as you were before.